### ISSUE TITLE
SC2086 with parameter expansion ':-' but not with ':+'

### ISSUE THREAD

**Issue by thranur** [koalaman/shellcheck#3189]
In the following script, the first echo gets an SC2086, but not the second one:

```sh
#!/bin/bash
echo ${VAR:-}
echo ${VAR:+$VAR}
```

I fail to see why that is the case; I thought that either both should be warned about, or none should.

In particular, if I run it with `VAR="a    b"` (with several spaces), in both cases, without double quotes, they echo `a b` (single space), and with double quotes, both echo `a    b`. So, for me, `${VAR:+$VAR}` should also deserve a SC2086.
